addons/point_of_sale/models/pos_session.py

```diff
@@ -300,15 +305,485 @@ def _validate_session(self):
+        for account_id in order_account_move_receivable_lines:
+            ( order_account_move_receivable_lines[account_id]
+            | invoice_receivable_lines[account_id]
+            ).reconcile()
+
+        # reconcile stock output lines
+        stock_moves = self.env['stock.move'].search([('picking_id', 'in', self.order_ids.mapped('picking_id').ids)])
+        stock_account_move_lines = self.env['account.move'].search([('stock_move_id', 'in', stock_moves.ids)]).mapped('line_ids')
+        for account_id in stock_output_lines:
+            ( stock_output_lines[account_id]
+            | stock_account_move_lines.filtered(lambda aml: aml.account_id == account_id)
+            ).reconcile()
+
+    def _prepare_line(self, order_line):
+        """ Derive from order_line the order date, income account, amount and taxes information.
+
+        These information will be used in accumulating the amounts for sales and tax lines.
+        """
+        def get_income_account(order_line):
+            product = order_line.product_id
+            income_account = product.property_account_income_id or product.categ_id.property_account_income_categ_id
+            if not income_account:
+                raise UserError(_('Please define income account for this product: "%s" (id:%d).')
+                                % (product.name, product.id))
+            return order_line.order_id.fiscal_position_id.map_account(income_account)
+
+        tax_ids = order_line.tax_ids_after_fiscal_position\
+                    .filtered(lambda t: t.company_id.id == order_line.order_id.company_id.id)
+        price = order_line.price_unit * (1 - (order_line.discount or 0.0) / 100.0)
+        taxes = tax_ids.compute_all(price_unit=price, quantity=order_line.qty, currency=self.currency_id, is_refund=order_line.qty<0).get('taxes', [])
+        date_order = order_line.order_id.date_order
+        taxes = [{'date_order': date_order, **tax} for tax in taxes]
+        return {
+            'date_order': order_line.order_id.date_order,
+            'income_account_id': get_income_account(order_line).id,
+            'amount': order_line.price_subtotal,
+            'taxes': taxes,
+        }
+
+    def _get_split_receivable_vals(self, payment, amount, amount_converted):
+        partial_vals = {
+            'account_id': payment.payment_method_id.receivable_account_id.id,
+            'move_id': self.move_id.id,
+            'partner_id': self.env["res.partner"]._find_accounting_partner(payment.partner_id).id,
+            'name': '%s - %s' % (self.name, payment.payment_method_id.name),
+        }
+        return self._debit_amounts(partial_vals, amount, amount_converted)
+
+    def _get_combine_receivable_vals(self, payment_method, amount, amount_converted):
+        partial_vals = {
+            'account_id': payment_method.receivable_account_id.id,
+            'move_id': self.move_id.id,
+            'name': '%s - %s' % (self.name, payment_method.name)
```

Comment:
shouldn't we take into account the partner when we are in combined, and split the combined by partner?

Reply:
This is the original implementation of the spec. Regardless of the customer, if the payment method is not split, then we them combine in single receivable line. But perhaps you're right. We need laura's opinion on this.